Gem/Transfer/ViewsModels/RecipientSceneViewModel.swift

```diff
@@ -8,54 +8,85 @@ import PrimitivesComponents
 import WalletService
 import Components
 import Style
-import SwiftUICore
+import NameResolver
+import Keystore
+import WalletsService
+import NodeService
+import SwiftUI
 
 typealias RecipientDataAction = ((RecipientData) -> Void)?
 
-enum RecipientAddressType: CaseIterable {
-    case pinned
-    case wallets
-    case view
-}
-
-enum RecipientScanResult {
-    case recipient(address: String, memo: String?, amount: String?)
-    case transferData(TransferData)
-}
+@Observable
+@MainActor
+final class RecipientSceneViewModel {
+    let keystore: any Keystore
+    let walletsService: WalletsService
+    let nodeService: NodeService
 
-class RecipientViewModel: ObservableObject {
     let wallet: Wallet
     let asset: Asset
     let type: RecipientAssetType
-    
+
     private let walletService: WalletService
+
     private let onRecipientDataAction: RecipientDataAction
     private let onTransferAction: TransferDataAction
     private let formatter = ValueFormatter(style: .full)
-    
+
+    var isPresentingScanner: RecipientScene.Field?
+    var nameResolveState: NameRecordState = .none
+    var memo: String = ""
+    var amount: String = ""
+    var addressInputModel: InputValidationViewModel
+
     init(
         wallet: Wallet,
         asset: Asset,
+        keystore: any Keystore,
         walletService: WalletService,
+        walletsService: WalletsService,
+        nodeService: NodeService,
         type: RecipientAssetType,
         onRecipientDataAction: RecipientDataAction,
         onTransferAction: TransferDataAction
     ) {
+
         self.wallet = wallet
         self.asset = asset
         self.walletService = walletService
+        self.walletsService = walletsService
+        self.keystore = keystore
+        self.nodeService = nodeService
+
         self.type = type
         self.onRecipientDataAction = onRecipientDataAction
         self.onTransferAction = onTransferAction
+
+        self.addressInputModel = InputValidationViewModel(
+            mode: .manual,
+            validators: [AddressValidator(asset: asset)]
```

Comment:
```suggestion
            validators: [AddressValidator(asset: asset)]
```

can you add extension so we can setup as `validators: [.required, .addressValidation(Chain), .minAmount(1)]` lomda ;ole tjos